recipes/gn/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+import textwrap
+import time
+
+required_conan_version = ">=1.33.0"
+
+
+class GnConan(ConanFile):
+    name = "gn"
+    description = "GN is a meta-build system that generates build files for Ninja."
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("gn", "build", "system", "ninja")
+    license = ("LGPL-3.0", "GPL-2.0")
+    homepage = "https://gn.googlesource.com/"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_compiler_version_supporting_cxx17(self):
+        return {
+            "Visual Studio": 15,
+            "gcc": 7,
+            "clang": 4,
+            "apple-clang": 10,
+        }.get(str(self.settings.compiler))
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
+        else:
+            if self._minimum_compiler_version_supporting_cxx17:
```

Comment:
I've seen this pattern over and over, I'm wondering if this should be a conan built in tool.
It would remove boilerplate code and eliminate the need from the recipe creator of knowing which c++ std supports each compiler.

Reply:
Some util functions would be nice.
But looking at the support matrix at https://en.cppreference.com/w/cpp/compiler_support,
the minimum required compiler version depends on what c++ features a project desires..

Reply:
-> https://github.com/conan-io/conan/issues/8002

Reply:
Thanks. One day... :smile:

Reply:
The C++ standard is not enough since the STL support changes between some minor versions it's a manual process for many projects